.github/workflows/buf-push.yaml

```diff
@@ -0,0 +1,33 @@
+name: buf-push
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'proto/**'
+
+jobs:
+  push:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+        # This step sets up the variable steps.buf-version.outputs.v
+        # to contain the version of Buf (e.g. v1.36.0) from go.mod.
+      - name: Get Buf version
+        id: buf-version
+        run: |
+          BUF_VERSION=$( go list -m -f '{{.Version}}' github.com/bufbuild/buf )
+          # remove the leading 'v' in front of the version
+          BUF_VERSION=${BUF_VERSION#v}
+          echo "v=$BUF_VERSION" >> "$GITHUB_OUTPUT"
+      - uses: bufbuild/buf-action@v1
+        with:
+          version: ${{ steps.buf-version.outputs.v }}
+          token: ${{ secrets.BUF_TOKEN }}
```

Comment:
So regarding the buf token in the validate action, try to execute this and you'll see the rate limiting errors:
```
$ for i in {1..100}; do make proto-generate; done
```

The error looks like this:

```
Failure: Unauthenticated remote plugin rate limit exceeded. For info on how to resolve this, visit: https://buf.build/b/auth-required
make: *** [proto-generate] Error 1
```

Currently it doesn't happen often, but according to Buf they will lower the limit until no unauthenticated requests are allowed.

Reply:
Oh and before doing that, execute `buf registry logout` to make sure you are not logged in.

Reply:
I see it now... It looks like logs I was looking for were actually coming from the buf GitHub actions not the tool itself. I'll fix this too, thanks for calling out.